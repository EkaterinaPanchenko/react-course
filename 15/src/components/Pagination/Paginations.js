import Pagination from 'react-js-pagination';
import { useDispatch, useSelector } from 'react-redux';

import { ACTION_CHANGE_PAGE, pageNumberSelector } from '../../ducks/paginationsReducer/paginationsReducer';
import { pageLimitSelector } from '../../ducks/paginationsReducer/paginationsReducer';
import { StyledPagination } from './style';

function Paginations({ totalItemsCount }) {
    
    const dispatch = useDispatch();
    const pageNumber = useSelector(pageNumberSelector);
    const pageLimit = useSelector(pageLimitSelector);
   
    const handlePageChange = (number) => {
        if(number !== pageNumber) {
            dispatch(ACTION_CHANGE_PAGE(number));
        }  
    };

    return(
            <StyledPagination>
                <Pagination
                    activePage={pageNumber}
                    itemsCountPerPage={pageLimit}
                    totalItemsCount={totalItemsCount}
                    pageRangeDisplayed={3}
                    onChange={handlePageChange}  
                />
            </StyledPagination>
    );
};

export default Paginations;