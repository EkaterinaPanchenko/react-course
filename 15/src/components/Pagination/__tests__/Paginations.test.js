import { configure, mount } from 'enzyme';
import { Provider } from 'react-redux';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import createMockStore from 'redux-mock-store';
import createSagaMiddleware from 'redux-saga';

import Paginations from '../Paginations';

configure({ adapter: new Adapter() });

const mockStore = createMockStore([createSagaMiddleware()]);

describe('<Paginations/> unit test', () => {
    const initialState = {
        ageOfEmpires: {
        fetchedData: { 
            civilizations: [],
            structures: [],
            technologies: [],
            units: [],
        },
        addData: [],
        isFetching: false,
        error: null 
        },
        paginations: {
            pageNumber: 1,
            offset: 0,
            pageLimit: 10
        }
    };

    const store = mockStore(initialState);
    const mockedTotalItemsCount = 20;
    const wrapper = mount(
        <Provider store={store}>
            <Paginations totalItemsCount={mockedTotalItemsCount}/>
        </Provider>
    );

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});