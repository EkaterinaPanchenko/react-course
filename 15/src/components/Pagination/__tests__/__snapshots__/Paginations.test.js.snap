// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paginations/> unit test should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Paginations
    totalItemsCount={20}
  >
    <styled.div>
      <div
        className="sc-dlfnbm MqQRM"
      >
        <Pagination
          activePage={1}
          firstPageText="«"
          getPageUrl={[Function]}
          hideFirstLastPages={false}
          innerClass="pagination"
          itemsCountPerPage={10}
          lastPageText="»"
          nextPageText="⟩"
          onChange={[Function]}
          pageRangeDisplayed={3}
          prevPageText="⟨"
          totalItemsCount={20}
        >
          <ul
            className="pagination"
          >
            <Page
              activeClass="active"
              ariaLabel="Go to first page"
              disabledClass="disabled"
              href="#"
              isActive={false}
              isDisabled={true}
              itemClass=""
              key="first"
              linkClass=""
              onClick={[Function]}
              pageNumber={1}
              pageText="«"
            >
              <li
                className="disabled"
                onClick={[Function]}
              >
                <a
                  aria-label="Go to first page"
                  className=""
                  href="#"
                >
                  «
                </a>
              </li>
            </Page>
            <Page
              activeClass="active"
              ariaLabel="Go to previous page"
              disabledClass="disabled"
              href="#"
              isActive={false}
              isDisabled={true}
              itemClass=""
              key="prev0"
              linkClass=""
              onClick={[Function]}
              pageNumber={0}
              pageText="⟨"
            >
              <li
                className="disabled"
                onClick={[Function]}
              >
                <a
                  aria-label="Go to previous page"
                  className=""
                  href="#"
                >
                  ⟨
                </a>
              </li>
            </Page>
            <Page
              activeClass="active"
              ariaLabel="Go to page number 1"
              disabledClass="disabled"
              href="#"
              isActive={true}
              isDisabled={false}
              key="1"
              onClick={[Function]}
              pageNumber={1}
              pageText="1"
            >
              <li
                className="active"
                onClick={[Function]}
              >
                <a
                  aria-label="Go to page number 1"
                  className="undefined"
                  href="#"
                >
                  1
                </a>
              </li>
            </Page>
            <Page
              activeClass="active"
              ariaLabel="Go to page number 2"
              disabledClass="disabled"
              href="#"
              isActive={false}
              isDisabled={false}
              key="2"
              onClick={[Function]}
              pageNumber={2}
              pageText="2"
            >
              <li
                className=""
                onClick={[Function]}
              >
                <a
                  aria-label="Go to page number 2"
                  className=""
                  href="#"
                >
                  2
                </a>
              </li>
            </Page>
            <Page
              activeClass="active"
              ariaLabel="Go to next page"
              disabledClass="disabled"
              href="#"
              isActive={false}
              isDisabled={false}
              itemClass=""
              key="next2"
              linkClass=""
              onClick={[Function]}
              pageNumber={2}
              pageText="⟩"
            >
              <li
                className=""
                onClick={[Function]}
              >
                <a
                  aria-label="Go to next page"
                  className=""
                  href="#"
                >
                  ⟩
                </a>
              </li>
            </Page>
            <Page
              activeClass="active"
              ariaLabel="Go to last page"
              disabledClass="disabled"
              href="#"
              isActive={false}
              isDisabled={false}
              itemClass=""
              key="last"
              linkClass=""
              onClick={[Function]}
              pageNumber={2}
              pageText="»"
            >
              <li
                className=""
                onClick={[Function]}
              >
                <a
                  aria-label="Go to last page"
                  className=""
                  href="#"
                >
                  »
                </a>
              </li>
            </Page>
          </ul>
        </Pagination>
      </div>
    </styled.div>
  </Paginations>
</Provider>
`;
